### PR TITLE
[feat] Create Accounts Page Outline

### LINKED ISSUE #59: Create My Account Page
https://linear.app/trap-garden/issue/TG-44/create-my-account-page

### PR DESCRIPTION
**Pull request by kaleenchen**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

**IMPORTANT NOTE:  Since this includes the changes from my first sprint and second sprint, go to commits tab and select the commit that is the latest (Adding Functionality to Accounts Page) to see the changes from first PR and this PR.**

[TASK 1: CREATING THE ACCOUNTS PAGE OUTLINE -- ALREADY REVIEWED BY KYLE & TORIN]
- Created the outline for MyAccounts page + styling + spacing
- Routing for logout button
- Grabbing user information from useProfile and Auth
- Handle logic for displaying plotOptions correctly

[TASK 2: ADDING FUNCTIONALITY TO ACCOUNTS PAGE -- NEEDS REVIEW]
- **linear task description**: https://linear.app/trap-garden/issue/TG-55/add-functionality-to-my-account-page
- **figma**: https://www.figma.com/design/M6GyVRxOPtLugZhLmQwuYF/TRAP-GARDEN-🪴?node-id=2481-16156&m=dev
- if user not signed in, app will reroute to login page
- if user hasn't completed onboarding -> button + text should appear under garden information, otherwise show user garden information 
- logout button positioning edited to be conditional based on has completed or not completed onboarding 

[NIT CHANGES FROM LAST PR FEEDBACK]
- px values converts to rm
- titleCase created for location and garden type
- back button added

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
Screenshot 2025-03-02 at 2 12 38 PM

[TASK 2: ADDING FUNCTIONALITY TO ACCOUNTS PAGE -- NEEDS REVIEW - PHOTOS]
Screenshot 2025-03-09 at 5 09 26 PM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- Review page.tsx for frontend logic
- review styles.ts for styling
- check spacing, colors, components 
- when logging out, should redirect to home page
- check handle logic for plotOptions [ie. if I need default option in line 141 if profileData doesn't return anything or leave empty]
- Aside: @kylezryr  sanity check the tsconfig bundler/ node

[TASK 2: ADDING FUNCTIONALITY TO ACCOUNTS PAGE -- NEEDS REVIEW]
- Since this includes the changes from my first sprint and second sprint, so to commits tab and select the commit that is the latest (Adding Funcationlity to Accounts Page) to see the changes from first PR and this PR.
- check that when user is NOT logged in, it is rerouted to login page
- back button has been added and has hydration error but @kylezryr said this is ok -- pls sanity check
- when user is not done with onboarding the My accounts page should not show any garden information and instead have a Go to Onboarding button and red text below. otherwise the rest of the garden info is shown
- log out button positioning updated conditional on if the user is onboarded or not 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: app/my-account/page.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  const router = useRouter();
+  const handleSignOut = async () => {
+    await signOut();
+    router.push(CONFIG.login);
+  };
+
+  return (
+    <Flex $direction="column">
+      {/* User profile image and name */}
+      <ProfilePictureContainer>
```
Comment (kylezryr) [inline app/my-account/page.tsx]: we're missing the back button here! you can copy the code from `/plant-page/all-plants`, pasted below:

```
         <BackButton
            onClick={() => {
              router.push(`/view-plants`); //you can replace this with router.back() 
            }}
          >
            ←
          </BackButton>
```

make sure the positioning matches the Figma!

File: app/my-account/page.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            marginTop: '16px',
+          }}
+        >
+          My Account
+        </H4>
+      </ProfilePictureContainer>
+
+      {/* Information regarding user personal information */}
+      <PersonalInformationContainer>
+        <H4
```
Comment (kylezryr) [inline app/my-account/page.tsx]: "Personal Information" and "Garden Information" are P1 tags according to the Figma!

File: app/my-account/page.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            Location
+          </P2>
+          <P2
+            style={{
+              marginBottom: '24px',
+            }}
+            $fontWeight={300}
+            $color={COLORS.darkgray}
+          >
+            {profileData?.us_state}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: we can call `toTitleCase` on `profileData?.us_state` so that it displays as title case instead of all upper case, as shown below. you can import `toTitleCase` from `utils/helpers.ts`.

we have to do the following to implement it:
```
 const { profileData, profileReady } = useProfile();

...
//inside your InfoField
            {profileReady && toTitleCase(profileData?.us_state!)}
```

this is because `profileData `takes some time to load in. we have a variable `profileReady `that turns true whenever the `profileData `is loaded in. `toTitleCase `expects a non-null value, so we can only call it on `profileData.us_state` when `profileReady `is true, so that the function receives a non-null value.

the exclamation mark after `profileData.us_state` asserts that the value is non-null, so that the function doesn't error. we know that it isn't null because it only renders when `profileReady `is true (due to the and statement) 

let me know if you need more explanation on this! 
image

File: app/my-account/page.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            Garden Type
+          </P2>
+          <P2
+            style={{
+              marginBottom: '24px',
+            }}
+            $fontWeight={300}
+            $color={COLORS.darkgray}
+          >
+            {profileData?.user_type}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: we can do something similar for the `user_type` so that it displays the corresponding user type but formatted correctly. we can do this by importing `userTypeLabels` from `/utils/helpers.ts` and calling `profileReady && userTypeLabels[profileData?.user_type as UserTypeEnum]` to be displayed here instead (you also have to import UserTypeEnum from `types/schema.d.ts`. 

this is because userTypeLabels expects a type of `UserTypeEnum` (which is a string that is either 'INDIV', 'SCHOOL' or 'ORG', but `profileData.user_type` is currently defined as a string, so we typecast it to a `UserTypeEnum`

File: app/my-account/page.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+          </P2>
+          <P2
+            style={{
+              marginBottom: '45px',
+            }}
+            $fontWeight={300}
+            $color={COLORS.darkgray}
+          >
+            {
+              plotOptions.find(option => option.value === profileData?.has_plot)
```
Comment (kylezryr) [inline app/my-account/page.tsx]: it might be easier to just hard-code this to say "No, I don't own a plot" if `profileData?.has_plot `is false, and vice versa, sorry about the confusion!

File: package.json [hunk 6]
```diff
@@ -36,5 +36,11 @@
     "prettier": "^3.3.3",
     "typescript": "5.4",
     "yarnhook": "^0.6.2"
+  },
+  "pnpm": {
+    "onlyBuiltDependencies": [
+      "sharp",
+      "supabase"
+    ]
```
Comment (kylezryr) [inline package.json]: we can remove this change from this file, it only happened because of the supabase/.env file error we had in the first worksesh!

File: tsconfig.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "bundler",
+    "moduleResolution": "node",
```
Comment (kylezryr) [inline tsconfig.json:11]: if adi said this is ok then it should be ok overall, I think we can leave this in

File: app/my-account/styles.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  flex-direction: column;
+  align-items: flex-start;
+  padding: 0 24px;
+`;
+
+export const InfoField = styled.section`
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+`;
```
Comment (kylezryr) [inline app/my-account/styles.ts:31]: could we convert the px values in this file to rem if possible? just divide px by 16 to get rem!

**Comment by kylezryr** [comment]
looks great overall! really great job with this PR! just a couple changes with styling and displaying the values from profileData!

File: app/my-account/page.tsx (hunk 5 above)
Comment (kaleenchen) [reply app/my-account/page.tsx]: Can you clarify this comment? Does line 141 not accomplish this already? (checking if profileData?.has plot is true or false and depending on which one it is its assigned to option and option.value is mapped to either the generic msg No.. or Yes..


I was going to hard code it until I found that the default messages are already in plotOptions

Comment (kylezryr) [reply app/my-account/page.tsx]: oh wait yeah that is actually better then, my bad you can leave it as is!

**Comment by torinnayak1** [comment]
Looks good! Yay kaleen goat

**Review by torinnayak1 — APPROVED** [review]
Looks great, the only thing that I saw was the dimensions flipping to rem but I believe Kyle already got that! Yayyyy

File: app/my-account/page.tsx [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+            paddingLeft: '1.5rem',
+            width: '100%',
+          }}
+        >
+          <BackButton
+            onClick={() => {
+              router.back();
+            }}
+          >
+            ←
```
Comment (kylezryr) [inline app/my-account/page.tsx]: we can replace this with the backArrow icon that got added to the icons file! you can do this by importing `Icon.tsx` from components and doing the following:

```
<BackButton ...>
    <Icon type='backArrow' />
</BackButton>
```

File: app/my-account/page.tsx [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+  return (
+    <Flex $direction="column">
+      {/* User profile image and name */}
+      <ProfilePictureContainer>
+        <div
+          style={{
+            display: 'flex',
+            paddingLeft: '1.5rem',
+            width: '100%',
+          }}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: you can replace this with a `Flex` container with the properties `$pl = '1.5rem'`

File: app/my-account/page.tsx [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+              flexDirection: 'column',
+              alignItems: 'center',
+              justifyContent: 'center',
+              width: '100%',
+              height: '100%',
+            }}
+          >
+            <BigButton
+              $primaryColor={COLORS.shrub}
+              onClick={() => router.push('/onboarding')}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: for consistency with the rest of the codebase, we store all the route names inside the `CONFIG` variable in `/lib/configs.ts`. you can import `CONFIG` and do `router.push(CONFIG.onboarding)` for this instead of hardcoding the route! this helps in case we change the onboarding route name later on

File: app/my-account/page.tsx [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+    <Flex $direction="column" $minH="calc(100vh - 60px)" $justify="between">
+      <div>
+        {/* User profile image and name */}
+        <ProfilePictureContainer>
+          <div
+            style={{
+              display: 'flex',
+              paddingLeft: '1.5rem',
+              width: '100%',
+            }}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: this can be a `Flex` container! you can check the source file of how `Flex` containers are defined to figure out which props you need to pass to replicate this div (check the `$pl$` prop!)

File: app/my-account/page.tsx [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+          {profileReady && !profileData ? (
+            <div
+              style={{
+                display: 'flex',
+                flexDirection: 'column',
+                alignItems: 'center',
+                justifyContent: 'center',
+                width: '100%',
+                height: '100%',
+              }}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: you can also make this a `Flex` container!

File: app/my-account/page.tsx [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+            >
+              <BigButton
+                $primaryColor={COLORS.shrub}
+                onClick={() => router.push('/onboarding')}
```
Comment (kylezryr) [inline app/my-account/page.tsx]: you can replace the `/onboarding` with `CONFIG.onboarding`, imported from `/lib/configs.ts`. we use this instead of hardcoding the routes so that in case we change the name of the onboarding route in the future, this is unaffected

File: app/my-account/page.tsx [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+  const handleSignOut = async () => {
+    await signOut();
+    router.push(CONFIG.login);
+  };
+
+  useEffect(() => {
+    if (!loading && !userEmail) {
+      router.push(CONFIG.login);
+    }
+  }, [userEmail, loading, router]);
```
Comment (kylezryr) [inline app/my-account/page.tsx:41]: works so far when not logged in!

**Review by kylezryr — CHANGES_REQUESTED** [review]
working great overall! styling is perfect, just to change a couple of divs to be `Flex`s instead for cleaner code, but functionality works well! 

i made some minor changes to change how the spacing works such that the log out button is always at the bottom of the screen! otherwise I left everything the same, great work overall!!

**Review by kylezryr — APPROVED** [review]
looks good to merge!
